New definitions lookup

lukemelia opened the pull request.

lukemelia committed:
WIP new definitions lookup

lukemelia committed:
DefintiionLookup implementation

lukemelia committed:
WIP use DefinitionLookup

ef4 committed:
configure --prerendererUrl on realm servers

ef4 committed:
cleanup test.only

lukemelia committed:
Include new test in realm server suite on CI

ef4 committed:
fix lints

ef4 committed:
Merge branch 'cs-9724-move-definitions-out-of-the-index' of github.com:cardstack/boxel-motion into cs-9724-move-definitions-out-of-the-index

ef4 committed:
fixing test setup type errors

ef4 committed:
More test setup updates

lukemelia committed:
caching def test registration fix

lukemelia committed:
Minor change to force CI update

lukemelia committed:
Merge branch 'main' into cs-9724-move-definitions-out-of-the-index

lukemelia committed:
Lint fix

lukemelia committed:
Fix module lookup key

lukemelia committed:
Definition lookup error handling adjustment

lukemelia committed:
Remove test no longer needed

lukemelia committed:
Refactor module definitions to include cache scope and auth user ID; update related tests and visibility handling

lukemelia committed:
Spec and lint fixes

lukemelia committed:
Use VirtualNetwork's fetch to probe remote realms for definition lookup

lukemelia committed:
Lint fix

lukemelia committed:
Merge branch 'main' into cs-9724-move-definitions-out-of-the-index

ef4 committed:
fix extension handling in definition cache

ef4 committed:
typo in start-base.sh

ef4 committed:
Merge remote-tracking branch 'origin/main' into cs-9724-move-definitions-out-of-the-index

lukemelia committed:
Update schema

ef4 commented on packages/realm-server/tests/definition-lookup-test.ts:
```diff
@@ -0,0 +1,258 @@
+      );
+    });
+
+    test('uses public cache scope when realm is public', async function (assert) {
```
Additional tests to add:
 - [x] same server public
 - [ ] cross server public
 - [ ] same server private
 - [ ] cross server private

ef4 commented on packages/runtime-common/definition-lookup.ts:
```diff
@@ -0,0 +1,426 @@
+  }
+}
+
+function parseJSON<T>(value: unknown, fallback: T): T {
```
Not needed (cast the db result type once, and no manual parsing should ever be required).

ef4 committed:
keep forRealm as part of the DefinitionLookup interface

lukemelia committed:
Update schema again

lukemelia committed:
Merge branch 'main' into cs-9724-move-definitions-out-of-the-index

lukemelia committed:
Tests: skip those not relevant to headless chrome indexing
Race condition: Guard lazilyLoadLinks flag per route

ef4 commented:
Status: we think the global lazilyLoadLinks flag might be leaking and causing an instance prerender to hit NotLoaded.

Debug logging branch is `move-definitions-debug-logging`.

lukemelia committed:
Merge branch 'main' into cs-9724-move-definitions-out-of-the-index

lukemelia committed:
Increase test timeout
Test reliability
Add test coverage
Test cleanup & make onInvalidation async
Try to get CI to pass for workspace-cli-sync

habdelra commented:
the prerendering-test flakiness should be fixed if you merge upstream

lukemelia committed:
Merge branch 'main' into cs-9724-move-definitions-out-of-the-index
